workshop/content/basics/operations/troubleshooting-provisioning-errors/_index.md

```diff
@@ -0,0 +1,116 @@
+
+To speed up development cycles, you can choose to [preserve](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/stack-failure-options.html) the state of resources that have been successfully provisioned as part of stack create and update operations, and of [change set](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/using-cfn-updating-stacks-changesets.html) operations. When you choose to use this functionality, you pause the stack rollback to preserve the state of successfully provisioned resources, and you can focus on troubleshooting and fixing your configurations, so that you can resume your provisioning operations when you are ready.
+
+### Topics Covered
+By the end of this lab, you will be able to:
+
+* Understand how to troubleshoot provisioning errors, whilst preserving the state resources you successfully deployed
+* Navigate the [AWS resource and property types reference](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-template-resource-type-ref.html) to discover resource properties, and return values for a given resource, from which you can choose as you develop your template
+
+### Start Lab
+You will use an example CloudFormation template, that contains an incorrect resource configuration, to create a new stack. You will choose to *preserve successfully provisioned resources* as part of *stack failure options* to preserve the state of an example `ExampleDeadLetterQueue` [Amazon SQS](https://aws.amazon.com/sqs/) queue resource, that will be successfully provisioned as part of the stack creation operation.
+
+The creation operation for your stack will fail because another SQS queue described in your template, `ExampleSourceQueue`, has a configuration error. You will troubleshoot and fix the error in the template, and then you will choose to resume the stack creation operation with the template you updated.
+
+To get started, follow steps shown next:
+1. Change directory to the `code/workspace/troubleshooting-provisioning-errors` directory.
+2. Open the `example_sqs_queues.yaml` CloudFormation template in your favorite text editor.
+3. Familiarize with the configuration for sample SQS queues in the template; your intents, in this example, are to:
+    1. create a *source* queue and a [*dead-letter* queue](https://docs.aws.amazon.com/AWSSimpleQueueService/latest/SQSDeveloperGuide/sqs-dead-letter-queues.html) (DLQ): you choose to have your `ExampleSourceQueue` target your `ExampleDeadLetterQueue` for messages that cannot be successfully processed.  In the template, you reference the [Amazon Resource Name](https://docs.aws.amazon.com/general/latest/gr/aws-arns-and-namespaces.html) (ARN) for  `ExampleDeadLetterQueue` in the [redrive policy](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-sqs-queues.html#aws-sqs-queue-redrive) for `ExampleSourceQueue`: your DLQ will then be created first, so that your *source* queue can then reference its ARN;
```

Comment:
```suggestion
    1. create a *source* queue and a [*dead-letter* queue](https://docs.aws.amazon.com/AWSSimpleQueueService/latest/SQSDeveloperGuide/sqs-dead-letter-queues.html) (DLQ): have your `ExampleSourceQueue` target your `ExampleDeadLetterQueue` for messages that cannot be successfully processed.  In the template, reference the [Amazon Resource Name](https://docs.aws.amazon.com/general/latest/gr/aws-arns-and-namespaces.html) (ARN) for  `ExampleDeadLetterQueue` in the [redrive policy](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-sqs-queues.html#aws-sqs-queue-redrive) for `ExampleSourceQueue`: your DLQ will then be created first, so that your *source* queue can then reference its ARN;
```